### ISSUE TITLE
Enhancement: Join search heads to an indexer cluster

### ISSUE THREAD

**Issue by derek126** [splunk/ansible-role-for-splunk#90]
We recently started using this project for our deployments.

We were able to successfully get our indexer cluster and search head cluster up and running using provided tasks, but noticed on the manager the search heads are not shown in the indexer clustering UI. Only the management node is shown under the search head.

I will happily attached our playbook and inventory if that would be helpful. It is pretty basic.

**Comment by derek126** [comment]
For those interested I added the following to the bottom of `configure_shc_members.yml`
```
- name: Integrate sh to indexer cluster
  command: "{{ splunk_home }}/bin/splunk edit cluster-config -mode searchhead -master_uri {{ splunk_uri_cm }} -secret {{ splunk_shc_key }} -auth {{ splunk_auth }}"
  become: true
  become_user: "{{ splunk_nix_user }}"
  register: shc_edit_result
  changed_when: shc_edit_result.rc == 0
  failed_when: shc_edit_result.rc != 0
  notify: restart splunk
  no_log: false 
  until: shc_edit_result.rc == 0
  retries: 6
  delay: 5
```
Works for me until an official solution or something.

**Comment by peterchenadded** [comment]
Above is not idempotent, you just need to create an ini_file task that loops through the 3 properties and store in system/local/server.conf and notify : restart Splunk.

**Comment by derek126** [comment]
We are using the role through Ansible galaxy pointing at this Git repo. Ideally it would be added to the role.

**Comment by peterchenadded** [comment]
Yeah, this repo is missing lots of features needed to actually run Splunk in prod.

**Comment by mason-splunk** [comment]
Presumably this would only be run once, when connecting the SH to the indexer cluster, so I'm not sure that idempotency is a concern. Using ini_file to loop through server.conf is also not idempotent because after the restart, splunk will replace the plaintext value of pass4SymmKey with a hash in server.conf. In the event that you are deploying a pre-hashed pass4SymmKey, then I would ask why you wouldn't just put the clustering config into a server.conf file and deploy it to the SH as app via `configure_apps.yml`.

**Comment by mason-splunk** [comment]
`configure_idxc_sh.yml` has been added to the v1.6.1 release to support this functionality.

**Comment by peterchenadded** [comment]
Hi @mason-splunk , it is idempotent if you have a task that can generically decrypt the encrypted value and compare it to the required value. If the values are different then it should update otherwise skip.

I've actually implemented this.

**Comment by derek126** [comment]
There is a a bug in configure_idxc_sh.yml. It should be
```
---
- name: Configure search head to join indexer cluster
  command: "{{ splunk_home }}/bin/splunk edit cluster-config -mode searchhead -master_uri {{ splunk_uri_cm }} -secret {{ splunk_idxc_key }} -auth {{ splunk_auth }}"
  become: true
  become_user: "{{ splunk_nix_user }}"
  register: idxc_sh_join_result
  changed_when: idxc_sh_join_result.rc == 0
  failed_when: idxc_sh_join_result.rc != 0
  notify: restart splunk
  no_log: false
  until: idxc_sh_join_result.rc == 0
  retries: 6
  delay: 5
```
Notice the `.rc`

**Comment by mason-splunk** [comment]
@derek126  Good catch. Fixed in v1.6.2 release.
